modules/drivers/snowflake/test/metabase/test/data/snowflake.clj

```diff
@@ -74,54 +119,91 @@
   []
   (sql-jdbc.conn/connection-details->spec :snowflake (tx/dbdef->connection-details :snowflake :server nil)))
 
-(defn- existing-dataset-names []
-  (let [db-spec (no-db-connection-spec)]
-    (jdbc/with-db-metadata [metadata db-spec]
-      ;; for whatever dumb reason the Snowflake JDBC driver always returns these as uppercase despite us making them
-      ;; all lower-case
-      (set (map u/lower-case-en (sql-jdbc.sync/get-catalogs metadata))))))
-
-(let [datasets (atom nil)]
-  (defn- existing-datasets []
-    (when-not (seq @datasets)
-      (reset! datasets (existing-dataset-names))
-      (log/infof "These Snowflake datasets have already been loaded:\n%s" (u/pprint-to-str (sort @datasets))))
-    @datasets)
-
-  (defn- add-existing-dataset! [database-name]
-    (swap! datasets conj database-name))
-
-  (defn- remove-existing-dataset! [database-name]
-    (swap! datasets disj database-name)))
+(defn- old-dataset-name?
+  "Is this dataset name prefixed by a date two days ago or older?"
+  [dataset-name]
+  (when-let [[_ year month day] (re-matches #"^(\d{4})_(\d{2})_(\d{2}).*$" dataset-name)]
+    (let [dataset-date (t/local-date (parse-long year) (parse-long month) (parse-long day))]
```

Comment:
should we add a `(try (catch ))` which returns true? If someone put February 31st accidentally we should consider it "old" and nuke it. currently it would just blow up every test runner which saw that database name.

Reply:
good point

Reply:
I'll add it

Reply:
added